### PR TITLE
Add pre/post upgrade test for Model Registry

### PR DESCRIPTION
**Pull request by lugi0**

### CODE REVIEW

**Review by tonyxrmdavidson — DISMISSED** [review]

File: ods_ci/tests/Resources/Page/ModelRegistry/ModelRegistry.resource [hunk 1]
```diff
@@ -0,0 +1,400 @@
...
+    ...    certs/${SECRET_PART_NAME_3}.key           certs/${SECRET_PART_NAME_3}.crt     certs/domain.crt
+    Secret Should Exist      ${NAMESPACE_ISTIO}    ${SECRET_PART_NAME_1}-credential
+    Secret Should Exist      ${NAMESPACE_ISTIO}    ${SECRET_PART_NAME_2}-credential
+    Secret Should Exist      ${NAMESPACE_MODEL_REGISTRY}  ${SECRET_PART_NAME_3}-credential
+
+Secret Should Exist
+    [Documentation]    Check if the specified secret exists in the given namespace
+    [Arguments]    ${namespace}    ${secret_name}
+    ${output}=    Run Process    oc get secret ${secret_name} -n ${namespace}    shell=True
+    Should Contain    ${output.stdout}    ${secret_name}
```
Comment (bdattoma) [inline ods_ci/tests/Resources/Page/ModelRegistry/ModelRegistry.resource:124]: you could use `Resource Should Exist`  from `ods_ci/tests/Resources/OCP.resource`

File: ods_ci/tests/Resources/Page/ModelRegistry/ModelRegistry.resource [hunk 2]
```diff
@@ -0,0 +1,400 @@
...
+Wait For Model Registry Containers To Be Ready
+    [Documentation]    Wait for model-registry-deployment to be ready
+    ${result}=    Run Process
+    ...        oc wait --for\=condition\=Available --timeout\=5m -n ${PRJ_TITLE} deployment/model-registry-db
+    ...        shell=true    stderr=STDOUT
+    Log To Console    ${result.stdout}
+    ${result}=    Run Process
+    ...        oc wait --for\=condition\=Available --timeout\=5m -n ${PRJ_TITLE} deployment/model-registry-deployment
+    ...        shell=true    stderr=STDOUT
+    Log To Console    ${result.stdout}
```
Comment (bdattoma) [inline ods_ci/tests/Resources/Page/ModelRegistry/ModelRegistry.resource:181]: maybe you could use `Wait For Pods Status` or `Wait For Deployment Replica To Be Ready` or `Wait For Pods To Be Ready`

File: ods_ci/tests/Resources/Page/ModelRegistry/ModelRegistry.resource [hunk 3]
```diff
@@ -0,0 +1,400 @@
...
+    IF    ${rc} > 0    Fail    Command 'oc whoami -t' returned non-zero exit code: ${rc}
+    ${token}=    Set Variable    ${token_result.stdout}
+    RETURN    ${token}
+
+Apply ServiceMeshMember Configuration
+    [Documentation]    Apply a ServiceMeshMember configuration using oc.
+    Copy File    source=${SERVICE_MESH_MEMBER}    destination=${MODELREGISTRY_BASE_FOLDER}/serviceMeshMember.yaml
+    Run    sed -i '' -e 's/<MODEL_REGISTRY_NS>/${NAMESPACE_MODEL_REGISTRY}/' ${MODELREGISTRY_BASE_FOLDER}/serviceMeshMember.yaml    # robocop: disable
+    Apply OpenShift Configuration    ${MODELREGISTRY_BASE_FOLDER}/serviceMeshMember.yaml
+    Remove File    ${MODELREGISTRY_BASE_FOLDER}/serviceMeshMember.yaml
```
Comment (bdattoma) [inline ods_ci/tests/Resources/Page/ModelRegistry/ModelRegistry.resource:212]: is this adding the MR namespace into the SMMR? if yes, we have a dedicated kw: `Add Namespace To ServiceMeshMemberRoll`

File: ods_ci/tests/Resources/Page/ModelRegistry/ModelRegistry.resource [hunk 4]
```diff
@@ -0,0 +1,400 @@
...
+    [Documentation]    Update the notebook
+    ${result}=    Run Process    python
+    ...    ${IPYNB_UPDATE_SCRIPT}    stdout=TRUE    stderr=TRUE
+    Log    ${result.stdout}
+    Log    ${result.stderr}
+    Should Contain    ${result.stdout}    Modified notebook saved
+
+Remove Model Registry
+    [Documentation]    Run multiple oc delete commands to remove model registry components
+    # We don't want to stop the teardown if any of these resources are not found
```
Comment (bdattoma) [inline ods_ci/tests/Resources/Page/ModelRegistry/ModelRegistry.resource:240]: the Run And Verify Command kw fails if the the command fails. 
In theory if this runs as teardown, it shouldn't matter

File: ods_ci/tests/Resources/Page/ModelRegistry/ModelRegistry.resource [hunk 5]
```diff
@@ -0,0 +1,400 @@
...
+    Upload Python Client Files In The Workbench    ${MR_PYTHON_CLIENT_FILES}
+    Upload Certificate To Jupyter Notebook    ${CERTS_DIRECTORY}/domain.crt
+    ${self_managed} =    Is RHODS Self-Managed
+    IF  ${self_managed}    Upload Certificate To Jupyter Notebook    openshift_ca.crt
+    Jupyter Notebook Can Query Model Registry     ${JUPYTER_NOTEBOOK}
+    SeleniumLibrary.Switch Window    ${handle}
+    Stop Workbench    workbench_title=${WORKBENCH_TITLE}
+    SeleniumLibrary.Close All Browsers
+
+Model Registry Post Upgrade Scenario
```
Comment (bdattoma) [inline ods_ci/tests/Resources/Page/ModelRegistry/ModelRegistry.resource:367]: why not checking that the service is still working from the workbench as well?

File: ods_ci/tests/Resources/Page/ModelRegistry/ModelRegistry.resource [hunk 6]
```diff
@@ -0,0 +1,400 @@
...
+    ...    oc delete project ${PRJ_TITLE} --force --grace-period=0
+    Should Be Equal As Integers	  ${return_code}	 0
+    Log    ${output}
+    ${NAMESPACE_MODEL_REGISTRY}=    Get Model Registry Namespace From DSC
+    Set Suite Variable    ${NAMESPACE_MODEL_REGISTRY}
+    Remove Model Registry
+    Remove Deployment Files    ${CERTS_DIRECTORY}
+    Remove Deployment Files    ${MODELREGISTRY_BASE_FOLDER}/Python_Dependencies
+    Disable Model Registry If Needed
+    RHOSi Teardown
```
Comment (bdattoma) [inline ods_ci/tests/Resources/Page/ModelRegistry/ModelRegistry.resource:400]: could you move these lines into a keyword to use as teardown of the post upgrade test? So if any of the previous checks fail, we are sure to clean the env

File: ods_ci/tests/Resources/Page/ModelRegistry/ModelRegistry.resource (hunk 2 above)
Comment (lugi0) [reply ods_ci/tests/Resources/Page/ModelRegistry/ModelRegistry.resource:181]: will be refactored later

File: ods_ci/tests/Resources/Page/ModelRegistry/ModelRegistry.resource (hunk 3 above)
Comment (lugi0) [reply ods_ci/tests/Resources/Page/ModelRegistry/ModelRegistry.resource:212]: No it's creating a SMM resource in the namespace, which has the effect of also adding the ns to the member roll

File: ods_ci/tests/Resources/Page/ModelRegistry/ModelRegistry.resource (hunk 5 above)
Comment (lugi0) [reply ods_ci/tests/Resources/Page/ModelRegistry/ModelRegistry.resource:367]: Tested manually for now so not strictly needed for 2.15, the whole test needs to be expanded but I want automation in place asap.

File: ods_ci/tests/Resources/Page/ModelRegistry/ModelRegistry.resource (hunk 1 above)
Comment (lugi0) [reply ods_ci/tests/Resources/Page/ModelRegistry/ModelRegistry.resource:124]: will be refactored in the future

**Review by tonyxrmdavidson — APPROVED** [review]
/lgtm

**Review by bdattoma — APPROVED** [review]

**Pull request merged**